### ISSUE TITLE
Assigning guards to towers doesn't include trainees

### ISSUE THREAD

**Issue by ravenbuilder934** [ldtteam/minecolonies#6026]
### Prerequisites 

- [ ] I am running the latest **alpha** version of MineColonies and Structurize for my Minecraft version.
- [x] I checked the MineColonies/Structurize [wiki](https://wiki.minecolonies.ldtteam.com/) and made sure my issue is not covered there.
- [x] I made sure that **this issue is not a duplicate of any existing issue.**

### Context 

- Minecraft Version: 1.15.2
- MineColonies Version: MineColonies Official pack v3.0
- Structurize Version: MineColonies Official pack v3.0
- Related Mods and their Versions: none

## Expected behavior
If I had the option in the guard/barracks tower GUI to recruit from combat academy/archery turned on, I would expect to see the combat academy/archery trainees in the list of hireable people.

## Actual behavior
Trainees do not show up in the list of hireable people, only the unemployed do.

## Steps to reproduce the issue

1. Hire at least one person at an Archery and have at least one unemployed colonist
2. Turn on 'recruit from combat academy/archery' in an Archer guard tower and set its hiring mode to Manual
3. Check the hireable people at the guard tower

### Logs

- latest.log: I noticed this on the patreon 1.15 server, unless you need clientside logs I can't help
- crashlog: 

### Notes
See https://github.com/ldtteam/minecolonies/issues/5842#issuecomment-718276118

---
## Viewers

* Add a thumbs-up to the bug report if you are also affected. This helps the bug report become more visible to the team and doesn't clutter the comments.
* Add a comment if you have any insights or background information that isn't already part of the conversation.

**Comment by Pyrocantaes** [comment]
Other than the fact I flagged this last Wednesday and was told it was missing functionality.. Applies to both Archers and Knights

**Comment by ravenbuilder934** [comment]
Right, I was just using archers/archer trainees as an example

**Comment by Raycoms** [comment]
Yeah it's not a bug, it's missing functionality. It's more of a feature request
